### ISSUE TITLE
yolov8 cls

### ISSUE THREAD

**Issue by lindsayshuo** [ultralytics/ultralytics#10008]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

If I need to train a multi label dataset, where an image has multiple attributes. For example, in an image, how to train the two labels of surprised facial expressions and wearing glasses

**Comment by lindsayshuo** [comment]
> @lindsayshuo hello! 😊
> 
> For training a multi-label dataset with YOLOv8, where an image can have multiple attributes like "surprised facial expressions" and "wearing glasses", you can set up your dataset in the following format:
> 
> 1. Each line in your label files (`*.txt`) corresponding to an image should list all the attributes present in that image. For example:
> 
> ```
> # In <image_name>.txt, assuming '0' corresponds to "surprised" and '1' to "wearing glasses":
> 0
> 1
> ```
> 
> 2. Your `dataset.yaml` should define all the classes (attributes) you have, for example:
> 
> ```yaml
> # In dataset.yaml
> train: /path/to/train/images
> val: /path/to/val/images
> 
> nc: 2  # number of classes
> names: ['surprised', 'wearing glasses']
> ```
> 
> 3. When training, set up your command as usual, pointing to your dataset and model configuration. YOLOv8 will treat each line in the label files as a separate attribute of the image.
> 
> Make sure your dataset is prepared correctly, with each image's attributes listed in their respective `.txt` files. This way, the model learns to recognize multiple attributes (labels) per image.
> 
> If you have any further questions or require more detailed guidance, feel free to ask!
> 
> Best of luck with your training! 🚀

data.yaml:

train: C:\Users\linds\Downloads\ultralytics-main\data_collected\train # 128 images
val: C:\Users\linds\Downloads\ultralytics-main\data_collected\train  # 128 images
test:
# number of classes
nc: 10

# class names
names: ['neutral', 'surprised', 'Sunglasses', 'fear', 'Glasses', 'anger', 'contempt', 'disgust', 'happy', 'sad']


default.yaml:

# Ultralytics YOLO 🚀, AGPL-3.0 license
# Default training settings and hyperparameters for medium-augmentation COCO training

task: classify # (str) YOLO task, i.e. detect, segment, classify, pose
mode: train # (str) YOLO mode, i.e. train, val, predict, export, track, benchmark

# Train settings -------------------------------------------------------------------------------------------------------
model: yolov8n-cls.yaml # (str, optional) path to model file, i.e. yolov8n.pt, yolov8n.yaml
data: data.yaml # (str, optional) path to data file, i.e. coco128.yaml
epochs: 100 # (int) number of epochs to train for
time: # (float, optional) number of hours to train for, overrides epochs if supplied
patience: 100 # (int) epochs to wait for no observable improvement for early stopping of training
batch: 16 # (int) number of images per batch (-1 for AutoBatch)
imgsz: 640 # (int | list) input images size as int for train and val modes, or list[w,h] for predict and export modes
save: True # (bool) save train checkpoints and predict results
save_period: -1 # (int) Save checkpoint every x epochs (disabled if < 1)
cache: False # (bool) True/ram, disk or False. Use cache for data loading


error:

Ultralytics YOLOv8.1.47 🚀 Python-3.11.5 torch-2.0.0+cu118 CUDA:0 (NVIDIA GeForce RTX 3060 Laptop GPU, 6144MiB)
engine\trainer: task=classify, mode=train, model=yolov8n-cls.yaml, data=data.yaml, epochs=100, time=None, patience=100, batch=16, imgsz=224, save=True, save_period=-1, cache=False, device=None, workers=8, project=None, name=train36, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, bgr=0.0, mosaic=1.0, mixup=0.0, copy_paste=0.0, auto_augment=randaugment, erasing=0.4, crop_fraction=1.0, cfg=None, tracker=botsort.yaml, save_dir=runs\classify\train36

Dataset not found ⚠️, missing path C:\迅雷下载\yolov5-master\yolov5-master\datasets\data.yaml, attempting download...
Downloading https://github.com/ultralytics/yolov5/releases/download/v1.0/data.yaml.zip to 'C:\迅雷下载\yolov5-master\yolov5-master\datasets\data.yaml.zip'...

Process finished with exit code -1

**Comment by glenn-jocher** [comment]
@lindsayshuo hi there! 👋 It looks like you're encountering a dataset path issue. Make sure the path in your `data.yaml` file is correctly pointing to where your training and validation datasets are located. The error you're getting indicates YOLOv8 can't find the dataset at the specified path.

Here's a quick suggestion: 

- Double-check the paths in your `data.yaml` file to ensure they correctly point to your dataset's location. Absolute paths are recommended for clarity.
- Your error log mentions an attempt to download a dataset from a URL which suggests the local path wasn't accessible. Make sure your dataset structure follows the expected format, and all files are in place.

If the paths are correct and the issue persists, consider if there are any restrictions or permissions affecting access to the dataset directory.

For example, ensure your `data.yaml` has accessible paths:

```yaml
train: /absolute/path/to/your/train/images
val: /absolute/path/to/your/val/images
```

Replace `/absolute/path/to/your/...` with the actual path to your datasets.

Hope this helps and happy training! 🚀

**Comment by lindsayshuo** [comment]
[    if str(dataset).startswith(("http:/", "https:/")):](https://github.com/ultralytics/ultralytics/blob/28f77d70116bfbe206861a0e1ded4ddf14c7b49d/ultralytics/data/utils.py#L368)

It seems that this function does not support loading yaml

**Comment by glenn-jocher** [comment]
Hi there! 👋 Thanks for spotting this! You're correct that the line you pointed out doesn't directly support loading `.yaml` from URLs. The primary intention there is to handle image or video sources from web URLs. 

However, for loading dataset configuration files like `data.yaml` from a URL, although not directly supported in that snippet, our intention is to provide flexibility elsewhere. 

For instance, if you need to load `data.yaml` from a URL, a workaround might involve downloading the file programmatically before passing it to your training or inference script. Here is a quick Python example on how you might do it:

```python
import requests

url = 'http://your-url.com/data.yaml'
r = requests.get(url)
with open('data.yaml', 'wb') as f:
    f.write(r.content)

# Now you can pass 'data.yaml' to your YOLO routine.
```

This can serve as an intermediate step to ensure your `data.yaml` is locally available for your training or inference sessions. Hope this helps! Let us know if you have any more questions. 🚀
